core/src/main/java/org/jboss/set/mjolnir/archive/GitDeleteRepository.java

```diff
@@ -0,0 +1,43 @@
+package org.jboss.set.mjolnir.archive;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.ListBranchCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.internal.storage.file.FileRepository;
+import org.eclipse.jgit.lib.Ref;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+public class GitDeleteRepository {
+
+    private String PATH;
+    private String gitUserName;
+
+    public GitDeleteRepository(String path, String gitUserName) {
+        this.PATH = path;
+        this.gitUserName = gitUserName;
+
+    }
+
+    public void gitRemoveBranches() throws GitAPIException, IOException {
+        File gitDir = new File(PATH);
+        List<Ref> refs = Git.open(gitDir)
+                .branchList().setListMode(ListBranchCommand.ListMode.REMOTE)
+                .call();
+        FileRepository localRepo = new FileRepository(PATH + "/.git");
+        Git git = new Git(localRepo);
+        refs.forEach(ref -> {
+            try {
+                if (ref.getName().contains(gitUserName)) {
```

Comment:
I would rather see this check to be a little stricter. If the ref names have the form "username/branchname" (?), then the test should better be something like `ref.getName().startsWith(gitUserName + "/")`.

Currently it could accept branches belonging to different users, if the branch name contained name of the user we are looking for. E.g. "thofman/lvydra-fixes" shouldn't pass when processing user "lvydra".

Reply:
ref.getName() giving ref/remote/gitUserName/gitbranch so here startsWith(gitUserName + "/") doesn't worked. instead I have done as if (ref.getName().contains(gitUserName + "/")) {  . Let me know if any othser changes required